Document pod DNS resolution schema (#20287)
kubernetes/website: content/en/docs/concepts/services-networking/dns-pod-service.md

## Pods

### A/AAAA records

> If pods created by associated `Deployment` or `DaemonSet` then the following
> schema of DNS resolution will be in place

Reviewer: ```suggestion
Any pods created by a `Deployment` or `DaemonSet` will have the following
DNS resolution available:
```

Reviewer 2: Actually can I suggest a further refinement:
```suggestion
Any pods created by a Deployment or DaemonSet have the following
DNS resolution available:
```

The [style guide](https://kubernetes.io/docs/contribute/style/style-guide/) recommends not putting resource names in backticks.

> `pod-ip-address.deployment-name.my-namespace.svc.cluster-domain.example.`

Reviewer 3: I could be wrong, but if you know the pod IP address what purpose does the DNS record serve?
```suggestion
`pod-name.deployment-name.my-namespace.svc.cluster-domain.example.`
```

Author: Couldn't agree more but I just documented the way how it currently works, changing it in docs will not fix it in general :) 

Would be great to have some insights from sig-network guys on existing implementation.

Reviewer: Hey @ytsarev and @jimangel - I pinged SIG Network in Slack and we’ll see if someone is able to audit this PR 👍🏼
